drivers/misc/sky13317/sky13317.c

```diff
@@ -79,9 +79,10 @@ int board_antenna_init(const struct device *dev)
 /**
  * Custom TI RFCC26XX callback for switching the on-board antenna mux on radio setup.
  */
-void board_cc13xx_rf_callback(RF_Handle client, RF_GlobalEvent events, void *arg)
+#if defined(CONFIG_SOC_CC1352P) || defined(CONFIG_SOC_CC1352P7)
```

Comment:
I think we still need a SoC specific antenna driver, but it should be able to use the device-tree settings to operate the antenna. The role of the SoC-specific driver is to connect the calls that switch the antenna (assigned through device-tree) to the SoC-specific calls on mode-changes that might require antenna switches. The Kconfig for custom HWattrs should enable the usage of the CC13x board antenna driver.